File: 2022/04/2.rb
# Original submission:
i = 0
File.readlines("input.txt").each do |line|
  a, b = line.split(',').map{|guy| guy.split('-').map(&:to_i)}
  a = (a[0]..a[1]).to_a
  b = (b[0]..b[1]).to_a

  if (a & b).any?
    i +=1
  end
end
p i



# Variation:
File.readlines("input.txt").count do |line|
  a, b = line.split(',').map{|guy| guy.split('-').map(&:to_i)}
  a = Range.new(*a).to_a
  b = Range.new(*b).to_a

  (a & b).any?
end.then {|cnt| p cnt}

File: 2022/04/2.py
import re

score = 0
with open('input.txt') as input:
    while line := input.readline():
        a, b, c, d = map(int, re.findall(r"\d+", line))
        left = range(a, b + 1)
        right = range(c, d + 1)

        if any(l in right for l in left) or any(r in left for r in right):
            score += 1

print(score)
